### ISSUE TITLE
make test fails with duplicate symbol

### ISSUE THREAD

**Issue by codeboten** [open-telemetry/opentelemetry-collector-contrib#5425]
**Describe the bug**
Running `make test` fails on MacOS arm64 with a duplicate symbol error.

```
make test
go test -race -timeout 60s --tags=containers_image_openpgp,exclude_graphdriver_btrfs,exclude_graphdriver_devicemapper ./...
?       github.com/open-telemetry/opentelemetry-collector-contrib/cmd/otelcontribcol    [no test files]
# github.com/open-telemetry/opentelemetry-collector-contrib/internal/components.test
/opt/homebrew/Cellar/go/1.17.1/libexec/pkg/tool/darwin_arm64/link: running clang failed: exit status 1
duplicate symbol '_readdrivestat' in:
    /var/folders/f2/5z5mvgcx6lx24rc_2_xzv2g40000gp/T/go-link-1335735787/000029.o
    /var/folders/f2/5z5mvgcx6lx24rc_2_xzv2g40000gp/T/go-link-1335735787/000032.o
ld: 1 duplicate symbol for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Steps to reproduce**
1. Clone the repo on a MacOS arm64 machine
2. Run `make test`

**What did you expect to see?**
The tests passing.

**What did you see instead?**
The error pasted above.

**What version did you use?**
This has been happening for some time, but I tested it last w/ `main` at version 1f91c2701d812bc7e7ab0ffa387d6999c96cef3b

**Environment**
OS: MacOS arm64
Compiler(if manually compiled): go 1.17.1

**Comment by mx-psi** [comment]
Reported on open-telemetry/opentelemetry-collector/discussions/4018. I think it makes more sense to keep this issue and close the other discussion since it seems to be a contrib-only problem.

**Comment by mx-psi** [comment]
After some `git bisect` it seems like it starts happening after #4386 (I am guessing the culprit is some dependency, maybe `golang.org/x/sys`?) and it only happens with Go 1.17 (which is why we didn't notice before).

**Comment by codeboten** [comment]
Can confirm this appears to be MacOS specific. I just ran the tests inside a Docker container on my mac and it appears to be fine:
```
root@14b3bb7009e0:/app# make test
go test -race -timeout 300s --tags=containers_image_openpgp,exclude_graphdriver_btrfs,exclude_graphdriver_devicemapper ./...
?   	github.com/open-telemetry/opentelemetry-collector-contrib/cmd/otelcontribcol	[no test files]
ok  	github.com/open-telemetry/opentelemetry-collector-contrib/internal/components	0.251s
ok  	github.com/open-telemetry/opentelemetry-collector-contrib/internal/version	0.020s [no tests to run]
```

**Comment by Aneurysm9** [comment]
It seems to be related to the https://github.com/shirou/gopsutil library used by the host metrics receiver.  It can be avoided by setting `CGOENABLED=0`, though I'm not sure that's the correct resolution.

**Comment by codeboten** [comment]
@mx-psi nice! I will give it a try later today.

**Comment by mx-psi** [comment]
@anuraaga shirou/gopsutil#1176 may solve the issues on the symbols mentioned on your last message. Would you mind checking if the issue still happens to you on main, and, if so, if the issue is fixed by bumping gopsutil to v3.21.11?

**Comment by anuraaga** [comment]
@mx-psi Thanks for the note, `make test` seems to work fine for me now on a Mac! Though lately I've switched to using remote development against a Linux VM so may not find future regressions :P
